lib/graphiti/scope.rb

```diff
@@ -17,6 +17,7 @@ def resolve
       if @query.zero_results?
         []
       else
+        @resource.adapter.apply_includes_on_scope(@object, sideload_name_arr)
```

Comment:
Sideload resolve results in multiple queries, To avoid it, we need to add `with_associations` on scope just before scope is resolved. Adding a handler here which will call method on adapter.

The `active_graph` adapter will apply `with_associations` when the handler is called ([ref here](https://github.com/mrhardikjoshi/graphiti-activegraph/blob/0b62124c406c640b7347f313afb5e8f7b11d9e1e/lib/graphiti/active_graph/adapters/active_graph.rb#L14-L16))